packages/plugins/src/hardhat/deployments.ts

```diff
@@ -336,23 +339,34 @@ export const proposeChugSplashBundle = async (
   configUri: string,
   remoteExecution: boolean,
   ipfsUrl: string,
-  isUpgrade: boolean,
-  configPath: string
+  configPath: string,
+  spinner: ora.Ora,
+  confirm: boolean
 ) => {
+  // Determine if the deployment is an upgrade
+  const isUpgrade = await checkIsUpgrade(hre.ethers.provider, parsedConfig)
   if (isUpgrade) {
+    spinner.succeed('Upgrade detected')
```

Comment:
I like that you use the spinner to inform the user whether the project is a deployment or upgrade, but there's a slight issue. In general, if you call `spinner.succeed` without calling `spinner.start` shortly after, there will be a pause in the spinner output until the next `spinner.start`. In this particular case, the pause might be pretty long since we commit the bundle to IPFS after this line. From the user's perspective, this could make it seem that the process is frozen when in fact it's not. In general, you can avoid this issue by calling `spinner.start` very shortly after each `spinner.succeed`. To demonstrate this, I made a PR [here]() that fixes this issue, since it was easier than leaving comments.